### ISSUE TITLE
[Bug]: "Set expiration date" can be disabled but it doesn't look disabled

### ISSUE THREAD

**Issue by thgoebel** [nextcloud/server#41171]
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on [Github](https://github.com/nextcloud/server/issues?q=is%3Aopen+is%3Aissue+label%3Abug) OR [Nextcloud Community Forum](https://help.nextcloud.com/) _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

The "Set expiration date" checkbox cannot be visually unchecked if a non-enforced default expiration date is configured by the admin.

This seems to have been introduced in https://github.com/nextcloud/server/pull/40629 in an attempt to fix https://github.com/nextcloud/server/issues/40612.

### Steps to reproduce

1. Admin: Enable "Set default expiration date for shares" and set a data but do NOT enable "Enforce expiration date" (under /settings/admin/sharing)
2. User: create a new share. Notice how since https://github.com/nextcloud/server/pull/40629 the default expiration date is correctly set
3.  Click "Set expiration date" in an attempt to un-check it

### Actual behaviour

The "Set expiration date" box remains checked. The date values are reset, and under the hood the expiration date is correctly unset. Just the UI is wrong.

### Expected behavior

The checkbox should become unchecked.

### Installation method

Community Docker image

### Nextcloud Server version

27.1.3

### Operating system

Debian/Ubuntu

### PHP engine version

PHP 8.2

### Web server

None

### Database engine version

None

### Is this bug present after an update or on a fresh install?

None

### Are you using the Nextcloud Server Encryption module?

Encryption is Disabled

### What user-backends are you using?

- [X] Default user-backend _(database)_
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

### Configuration report

```shell
n/a, see https://github.com/nextcloud/server/issues/40612
```

### List of activated Apps

```shell
n/a, see https://github.com/nextcloud/server/issues/40612
```

### Nextcloud Signing status

```shell
No errors have been found.
```

### Nextcloud Logs

```shell
n/a
```

### Screenshot

<img src=https://github.com/nextcloud/server/assets/33295590/65a3e56c-b635-4cb8-bfca-0334d555c55a width=65%>

**Comment by szaimen** [comment]
Cc @Fenn-CS

**Comment by 42ske** [comment]
Sorry, I don't see if this bug should already be fixed in NC 27.1.3 (for me it still exists in this release) or will be fixed with 27.1.4?
